packages/back-end/src/controllers/experiments.ts

```diff
@@ -1632,13 +1631,14 @@ export async function addScreenshot(
 
   experiment.variations[variation].screenshots =
     experiment.variations[variation].screenshots || [];
-  experiment.variations[variation].screenshots.push({
+
+  changes.variations = experiment.variations;
+
+  changes.variations[variation].screenshots.push({
```

Comment:
This will still mutate the original.  You need to deep clone `experiment.variations` first before modifying it.

Reply:
@jdorn Got this updated. Let's talk through this at our 1:1 today as I don't fully grok why the `deepClone` is necessary here as we don't appear to be using the original `experiment` object (other than it's `id`) or returning it later in this method.

Additionally, when looking at the `deleteScreenshot` method above, it looks like we are doing something similar, but not using cloneDeep there - we're directly mutating the `experiment` object via the use of the `.filter` method, before saving `changes.variations = experiment.variations`. Should we also be leveraging `cloneDeep` there?

If not, let's talk through why.